server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -303,11 +307,23 @@ public void validate(final TimeValue value, final Map<Setting<?>, Object> settin
             final Boolean fastRefresh = (Boolean) settings.get(INDEX_FAST_REFRESH_SETTING);
             final IndexVersion indexVersion = (IndexVersion) settings.get(SETTING_INDEX_VERSION_CREATED);
 
+            final boolean allowOverride = Boolean.parseBoolean(
+                System.getProperty(STATELESS_ALLOW_INDEX_REFRESH_INTERVAL_OVERRIDE, "false")
+            );
```

Comment:
Can we parse this into an instance variable instead to avoid parsing this boolean on every validation (which does run somewhat frequently).

Reply:
Sure. Reworked it to a class variable (since the flag value is not going to change during node's lifetime) also made init in a "lazy" way to facilitate unit testing (our testing framework does not allow to control well when static init happens)

Reply:
I think making the parsing static should be sufficient which is the common practice for other system properties. Using `LazyInitializable` has unnecessary cost of reading a volatile field.